Cargo.toml

```diff
@@ -18,6 +18,7 @@ minimp3_fixed = { version = "0.5.4", optional = true}
 symphonia = { version = "0.5.4", optional = true, default-features = false }
 crossbeam-channel = { version = "0.5.8", optional = true }
 thiserror = "1.0.49"
+rand = {version = "0.8.5", features = ["small_rng"]}
```

Comment:
we do not want to depend rand if the noise waveform functions are not needed by the user. So rand should be an optional dependency enabled by some compile time feature. Lets call that feature `noise`?

Reply:
Is there a place for feature documentation?

Reply:
not yet, but that is a good idea :+1: 

best place would probably be here: https://github.com/RustAudio/rodio/blob/master/src/lib.rs

I am planning to overhaul the features related to decoders later so for now lets only document this one.